### ISSUE TITLE
Required variables broken in 0.49.0

### ISSUE THREAD

**Issue by urkle** [metabase/metabase#40250]
### Describe the bug

Our devops upgraded our metabase last week to 0.49.0 and since then any report that has a required field (but no default) will not run!.  No matter what we set the value to in the input fields it *thinks* the field is not set and refuses to run the report.

We have to toggle the require property of the variable and then the report will run. 

### To Reproduce

1. Have a  SQL report with a required parameter variable
2. have that variable be required, but with no default value.
3. specify a value for the property
4. the "run" icon is NOT clickable (remains disabled)

### Expected behavior

To be able to run the report as in previous metabase versions.

### Information about your Metabase installation

```JSON
- browser - Firefox 123.0.1
- browser OS macOS 12
- DBs for queries (doesnt matter) redshfit & postgresql
- Metabase 0.49.0
- hosting AWS
- internal DB postgresql
```

### Severity

annoying and semi-blocking

**Comment by paoliniluis** [comment]
We need to force the user to set up a default value here

**Comment by oleggromov** [comment]
@urkle hi, the author of this new feature here. I am sorry that you've experienced a regression! 
To help us resolve the issue quickly, could you please confirm that you are blocked by what's explained on the screenshot?

<img width="1280" alt="Screenshot 2024-03-18 at 18 54 50" src="https://github.com/metabase/metabase/assets/2196347/13e6aaaf-ebc4-4e45-a820-87d229f122ce">

**Comment by urkle** [comment]
@oleggromov yes, that is what we are encountering (in sql query mode).

**Comment by oleggromov** [comment]
Hi @urkle the problem was fixed, please expect it to be shipped with `0.49.1` version, which is due on March 28.

**Comment by darksciencebase** [comment]
@urkle we've actually just released 49.1, a bit ahead of the schedule.

**Comment by cbalusek** [comment]
This issue is not resolved in 49.1, but in 49.3, which will be released early next week

**Comment by urkle** [comment]
Thanks @cbalusek I noticed the issue was only partially fixed in 0.49.1 (I could run them but not save them)

**Comment by danielcarletti** [comment]
With this fix, every question with mandatory parameters now also requires a default value. Before, we could leave it empty so the user had to fill in the parameter. Is that an intended change or an unwanted side effect?
